packages/runtime/src/internal/data/resolveData.ts

```
@@ -53,24 +62,35 @@ export async function resolveData(
 
-  const [providerBrick, actualArgs] = await Promise.all([
+  const contractConf =
+    !Array.isArray(args) &&
+    params &&
+    hooks?.flowApi?.isFlowApiProvider(useProvider)
+      ? ({ params, options, filename } as UserProviderContractConf)
+      : null;
+
+  const [providerBrick, actualArgs, actualContractConf] = await Promise.all([
     getProviderBrick(useProvider) as unknown as Promise<
       Record<string, Function>
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)